server/src/main/java/org/elasticsearch/common/blobstore/BlobContainer.java

```diff
@@ -74,6 +74,28 @@
      */
     void writeBlob(String blobName, InputStream inputStream, long blobSize) throws IOException;
 
+    /**
+     * Reads blob content from the input stream and writes it to the container in a new blob with the given name,
+     * using an atomic write operation if then implementation supports it. When atomic writes are not supported,
```

Comment:
s/then/the/
Also it's rather the opposite. If the implementation support atomic writes by default, it then delegates to writeBlob...?

Reply:
"When the BlobContainer implementation does not provide a specific implementation for wireBlobAtomic, then the usual writeBlob method is used." Would that be better?

Reply:
yes